titus-common/src/test/java/com/netflix/titus/common/util/rx/ReactorReEmitterOperatorTest.java

```diff
@@ -17,69 +17,75 @@
 package com.netflix.titus.common.util.rx;
 
 import java.time.Duration;
-import java.util.concurrent.TimeUnit;
 
 import org.junit.Test;
-import reactor.core.publisher.DirectProcessor;
 import reactor.core.publisher.Flux;
 import reactor.core.scheduler.Schedulers;
 import reactor.test.StepVerifier;
 
+import static org.assertj.core.api.Assertions.assertThat;
+
 public class ReactorReEmitterOperatorTest {
 
-    private static final long INTERVAL_MS = 1_000;
+    private static final Duration EMIT_INTERVAL = Duration.ofMillis(1_000);
 
-    private final DirectProcessor<String> subject = DirectProcessor.create();
+    private boolean cancelled;
```

Comment:
given we are using `StepVerifier` and `Schedulers.parallel()`, it would probably be safer to make this `volatile` to avoid flakiness.

Reply:
`StepVerifier` swaps `Schedulers.parallel()` with test scheduler, which makes it effectively run on the main thread.